src/main/java/com/datadoghq/sketch/ddsketch/store/DenseStore.java

```diff
@@ -47,10 +47,17 @@
     DenseStore(DenseStore store) {
         this.arrayLengthGrowthIncrement = store.arrayLengthGrowthIncrement;
         this.arrayLengthOverhead = store.arrayLengthOverhead;
-        this.counts = store.counts == null || store.isEmpty() ? null : Arrays.copyOf(store.counts, store.counts.length);
-        this.offset = store.offset;
         this.minIndex = store.minIndex;
         this.maxIndex = store.maxIndex;
+        if (store.counts != null && !store.isEmpty()) {
+            this.counts = Arrays.copyOfRange(store.counts,
+                    store.minIndex - store.offset,
+                    store.maxIndex - store.offset + 1);
+            this.offset = 0;
```

Comment:
Shouldn't it be `this.offset = store.minIndex`? If I remember correctly, the `offset` is the index of the first element of the `counts` array.

Reply:
No, because as part of the trimming process I removed the empty values at the start of the array.

Reply:
That's not what the `offset` represents. We use it to turn the initial index into the `count` array index (see [this](https://github.com/DataDog/sketches-java/blob/70d60fda26b51a98292f873d328a55ad1c836340/src/main/java/com/datadoghq/sketch/ddsketch/store/UnboundedSizeDenseStore.java#L33) for instance).

This code:

```
final UnboundedSizeDenseStore store = new UnboundedSizeDenseStore();
store.add(10);
System.out.println(store.getTotalCount());
System.out.println(store.copy().getTotalCount());
```

generates:

```
1.0
0.0
```

I think we can easily add tests for the copy by testing the copy in `StoreTest.test` as we test the serialized/deserialized sketch.

Reply:
OK. I can address this. Alternatively, we could leave this optimisation to another time because I don't plan on using the dense store anyway, and it's a nice to have to free up some memory which wasn't freed up before.

Reply:
Yes, it's fine with me either way. I can take care of it in a separate PR if you prefer.

Reply:
I've addressed this and added a test for all store types that counts are preserved after a copy.